src/uTensor/ops/Reshape.hpp

```diff
@@ -0,0 +1,64 @@
+#ifndef UTENSOR_RESHAPE_H
+#define UTENSOR_RESHAPE_H
+
+#include "context.hpp"
+#include "types.hpp"
+#include "tensor.hpp"
+#include "uTensor_util.hpp"
+#include "operatorBase.hpp"
+
+using std::array;
+
+namespace uTensor {
+
+template <typename Tin>
+class ReshapeOperator : public OperatorInterface<1, 1> {
+/* reshape input as the shape of output*/
+public:
+  ReshapeOperator(const TensorShape&& shape) : _shape(shape) {}
+  ReshapeOperator(const TensorShape& shape) : _shape(shape) {}
+  enum names_in : uint8_t { input };
+  enum names_out : uint8_t { output };
+  virtual void compute(){
+    const Tensor& input_tensor = inputs[input].tensor();
+    Tensor& output_tensor = outputs[output].tensor();
+    output_tensor->resize(_shape);
+    if (input_tensor->num_elems() != output_tensor->num_elems()){
```

Comment:
Isnt this a valid use case? I thought resize could shrink and delete elements?

Reply:
Yes, resize could add or delete elements.
But **reshape** should keep the number of elements.

Reply:
ah you are right, I misread